ios/RNCWebView.m

```diff
@@ -238,6 +225,9 @@ - (void)didMoveToWindow
           [script appendFormat:@"document.cookie = %@ + '=' + %@",
             RCTJSONStringify(cookie.name, NULL),
             RCTJSONStringify(cookie.value, NULL)];
+          if (cookie.domain) {
+            [script appendFormat:@" + '; Domain=' + %@", RCTJSONStringify(cookie.domain, NULL)];
```

Comment:
Shouldn't we check that the cookie domian is the same than the request one?

Reply:
Good question.
I've added this snippet when retrieving cookies, before even building the script:

```
NSURLRequest *request = [RCTConvert NSURLRequest:self.source];

NSArray<NSHTTPCookie *> *cookies = (request != nil && request.URL != nil) ?
            [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookiesForURL:request.URL]
            : [[NSHTTPCookieStorage sharedHTTPCookieStorage] cookies];
```

Which is pretty much like the handling from the `requestForSource:` method: If the `WebView` needs to load an URL, convert the JSON passed from RN as params to a `NSURLRequest` object then check its domain.